libs/sdk-ui-dashboard/src/model/store/storeAccessors/DashboardStoreAccessorRepository.ts

```diff
@@ -0,0 +1,129 @@
+import { ObjRef } from "@gooddata/sdk-model";
+
+/**
+ * This singleton class serves the selector and the dispatcher properties for given dashboard {@link @gooddata/sdk-model#IdentifierRef}.
+ *
+ * The {@link Dashboard} component has an optional property {@link IDashboardProps#onStateChange} through which
+ * you can handle setting of the values for {@link DashboardDispatch} and {@link DashboardSelectorEvaluator}.
+ *
+ *
+ * In your component using {@link Dashboard} you can create an onStateChange callback for your dashboard using
+ * {@link DashboardStoreAccessorRepository#getOnChangeHandlerForDashboard} and pass it to mentioned Dashboard
+ * component property.
+ *
+ * @example
+ * ```
+ *  const dashboardStoreAccessor = MultipleDashboardStoreAccessor.getInstance();
+ *
+ *
+ *  // in the code where needed
+ *  dashboardStoreAccessor.getAccessorsForDashboard(<DASHBOARD_ID>).getDispatch()(
+ *      changeDateFilterSelection("relative", "GDC.time.month", "-3", "0"),
+ *  );
+ *
+ *  // or with check if accessor is initialized already
+ *  if (dashboardStoreAccessor.isAccessorInitializedForDashboard(DASHBOARD_ID)) {
+ *          setSelectResult(
+ *              dashboardStoreAccessor.getAccessorsForDashboard(DASHBOARD_ID).getSelector()(
+ *                  selectEffectiveDateFilterOptions,
+ *              ),
+ *          );
+ *      }
+ *
+ *  return (
+ *      <Dashboard dashboard={"<dashboardId>"} onStateChange={dashboardStoreAccessor.getOnChangeHandlerForDashboard(DASHBOARD_ID)}/>
+ *  )
+ * ```
+ *
+ * To get latest properties, use static member function {@link DashboardStoreAccessor#getInstance}. If there is already an instance
+ * created, it will return this instance and will return new instance of the {@link DashboardStoreAccessor} otherwise.
+ *
+ * @public
+ */
+export class DashboardStoreAccessorRepository {
+    static dashboardAccessor: DashboardStoreAccessorRepository;
+
+    accessors = new Map<ObjRef, DashboardStoreAccessor>();
+
+    private constructor() {
+        // empty on purpose to prevent incorrect use of this class
+    }
+
+    /**
+     * {@link DashboardStoreAccessorRepository#getInstance} is the only correct way to get the instance of
+     * the {@link DashboardStoreAccessorRepository} class.
+     *
+     * @Returns the existing {@link DashboardStoreAccessorRepository} or create and return new one.
+     */
+    static getInstance() {
+        if (!DashboardStoreAccessorRepository.dashboardAccessor) {
+            DashboardStoreAccessorRepository.dashboardAccessor = new DashboardStoreAccessorRepository();
+        }
+        return this.dashboardAccessor;
+    }
+
+    /**
+     * Gets the correct {@link DashboardStoreAccessor} for given dashboard from the {@link DashboardStoreAccessorRepository#accessors} map.
+     *
+     * @param dashboard - an {@link @gooddata/sdk-model#IdentifierRef} of the dashboard.
+     */
+    getAccessorsForDashboard(dashboard: ObjRef): DashboardStoreAccessor {
```

Comment:
We could probably extend the API here so that it is in line with the Dashboard component: https://github.com/gooddata/gooddata-ui-sdk/blob/master/libs/sdk-ui-dashboard/src/presentation/dashboard/Dashboard.tsx#L220-L222 so that you can pass either an `ObjRef` (then it is used as-is) or a string (then it is converted to idRef).

This is applicable to the other methods here as well.